feat: Admin Console Development (1st dev)

yoongyu-lee opened the pull request:
## Description
This pull request introduces the Admin Console built with React, integrated into the existing backend server. The frontend project is designed to be built and copied into the `resource/static` directory of the backend. The backend source is restructured to include admin and agent packages, where the admin package contains the Admin Console server logic. Initial data is also configured to support integration with the existing demo server.

## Related Issue (Optional)
- Issue #

## Changes
- Developed Admin Console with the following features:
  - VC Schema Management
  - VC Plan Management
  - Notification Provider Settings
  - KYC Settings
  - List Provider Settings
  - Admin Account Management
- Integrated frontend into backend's static resource directory after build
- Removed hardcoded logic and sample data from TA Server:
  - Eliminated hardcoded YAML configuration values
  - Removed embedded sample data

## Screenshots (Optional)
<img width="807" alt="2-1 main-screen" src="https://github.com/user-attachments/assets/83a8f551-07d2-43ab-9d18-d907b648874c" />

yoongyu-lee committed:
chore(gitignore): add ignore rules for React and Visual Studio Code

yoongyu-lee committed:
Initial commit (ta-admin-frontend)

yoongyu-lee committed:
refactor(tas): Restructure package hierarchy

- Moved all sources under `tas.v1` to `tas.v1.agent`
- Created `tas.v1.admin` package
- Created `tas.v1.common` package

yoongyu-lee committed:
refactor(tas): move query package under common

The simple query service needs to be used in the admin as well, so it has been relocated.

yoongyu-lee committed:
feat(frontend): develop TA registration page

yoongyu-lee committed:
feat(frontend): modify proxy configuration (api -> admin/v1)

yoongyu-lee committed:
feat(frontend): remove custom BreadcrumbsGenerator

yoongyu-lee committed:
feat(frontend): remove '/' mapping from router configuration (fix breadcrumbs 1-depth issue)

yoongyu-lee committed:
feat: add loading component and server status context

- Added `LoadingScreen` component for better UX during API calls
- Introduced `ServerStatusContext` to manage and provide server status to child components
- Removed unused theme files for cleaner codebase
- Updated TypeScript file naming convention to start with lowercase

yoongyu-lee committed:
fix: remove tsconfig.tsbuildinfo file

yoongyu-lee committed:
feat(gitignore): ignore tsconfig.tsbuildinfo file

yoongyu-lee committed:
feat(frontend): add navigationConfig

yoongyu-lee committed:
fix: prevent infinite loading on server failure and replace login API with health check API

Temporarily replaced the login API with the health check API until the login API is developed.
Fixed an issue where the application would enter an infinite loading state if the server connection failed.
Started development of the TA registration page.

yoongyu-lee committed:
feat(backend): developing TA registration API

yoongyu-lee committed:
feat: developing TA registration feature - Successfully issued issuing certificate

yoongyu-lee committed:
feat(ta-registration): complete TA registration and retrieval development

yoongyu-lee committed:
feat: Developing entity management page

yoongyu-lee committed:
feat(entities): Completed entity list retrieval, registration, and quick registration

yoongyu-lee committed:
feat: show confirmation dialog when clicking the "Quick Register Entities" button

yoongyu-lee committed:
feat: update status check logic when entering TA Management and TA Registration pages

yoongyu-lee committed:
feat(entities): Refactor entity registration validation code.

yoongyu-lee committed:
feat(entities): add logic to send the certificate vc to the entity after simple entity registration

yoongyu-lee committed:
feat: allow external absolute path injection for blockchain property file

1. Moved BaseBlockChainUtil functionality to BlockChainServiceImpl
2. Managed blockchain.properties file via yml configuration
3. Applied did-blockchain-sdk-server-1.0.0.jar to temporarily handle absolute path for property file

yoongyu-lee committed:
refactor: rename file-related property keys and update related classes

yoongyu-lee committed:
feat(frontend): ensure setIsLoading(false) is called in catch block

yoongyu-lee committed:
feat(kyc): developing KYC configuration Page

yoongyu-lee committed:
feat(kyc): Completed KYC configuration

yoongyu-lee committed:
refactor: move DTO package from common to agent and admin

- Moved DTOs from the common package to the agent and admin sub-packages
- Updated comments accordingly

yoongyu-lee committed:
feat(api-config) : Developing API configuration, Add CustomAccount, Ad API liquibase xml

yoongyu-lee committed:
feat(api): Completed Expiration Settings configuration

yoongyu-lee committed:
feat(api): Completed Key Exchange Policy Settings configuration

yoongyu-lee committed:
feat(api): disable Update button when values are unchanged

yoongyu-lee committed:
fix: update 'Entity Simple Installation' logic due to setup path change

yoongyu-lee committed:
fix: remove unused KycFeign class

yoongyu-lee committed:
fix: change Orchestrator file path (ta.did)

yoongyu-lee committed:
chore(gitignore): add .vite/

yoongyu-lee committed:
feat: Merged OpenDidAdminException into OpenDidException

yoongyu-lee committed:
refactor: Split noti package into agent and admin packages

yoongyu-lee committed:
feat(noti): Add Liquibase file for notification_server table

yoongyu-lee committed:
feat(noti-email-server): Completed Email Server Settings

yoongyu-lee committed:
feat(noti-email-server): Completed Email Template Settings

yoongyu-lee committed:
feat(login): Complete login

In the second phase of development, the backend server will issue a login token.

yoongyu-lee committed:
feat(noti-push-server): Completed Push Server Settings

yoongyu-lee committed:
feat: change API file name (frontend), fix paging bug, modify TA registration page design

yoongyu-lee committed:
rerefactor: separate noti package into agent and admin (backend), rename API TypeScript file, developing allowed CA list feature

yoongyu-lee committed:
feat(allowed-ca-list): Completed to develop Allowed CA List feature, Applied Common SDK 1.0.1-SNAPSHOT

yoongyu-lee committed:
feat(db): make wallet_id column to unique

yoongyu-lee committed:
feat(service): Completed VC Schema feature (1st dev), fix initializeFirebase bug (add Exception)

yoongyu-lee committed:
feat(vc-shema): Developing TA simple registration using vc schema table

yoongyu-lee committed:
feat(service): Completed VC Plan feature (1st dev)

yoongyu-lee committed:
feat(service): Completed Admin feature (1st dev)

yoongyu-lee committed:
feat(admin): apply admin permissions to Admin main page

yoongyu-lee committed:
refector(FileLoaderService): Remove FileLoaderService class

yoongyu-lee committed:
refactor(FcmProperty): Remove FcmProperty class

yoongyu-lee committed:
feat: Change Entity server health check URL to domain:port/actuator/health

yoongyu-lee committed:
feat: Completed to develop change password (self)

yoongyu-lee committed:
fix(email-server-setting): fix bug where email server settings couldn't be registered if user didn't change the select box (e.g: SSL Enabled)

yoongyu-lee committed:
fix(kyc): apply baseUrl for ping test

yoongyu-lee committed:
fix(index): change the title

yoongyu-lee committed:
refactor(error): Add errorHandler utility for consistent error handling

yoongyu-lee committed:
refactor: change file name from errorHandler.ts to error-handler.ts

yoongyu-lee committed:
refactor(error): Add sha256-hash utility

yoongyu-lee committed:
fix: add missing import for SelectChangeEvent

yoongyu-lee committed:
feat(frontend): change back button to cancel button, Make the Popover component responsive

yoongyu-lee committed:
feat: update subtitle on sign-in page and add dashboard page

yoongyu-lee committed:
feat(degisn): applying new design updates

yoongyu-lee committed:
feat(degisn): applying new design updates (90%)

yoongyu-lee committed:
feat(degisn): applying new design updates (100%)

yoongyu-lee committed:
feat(frontend): Change dialog button position and update error button color

yoongyu-lee committed:
feat(frontend): change button position

yoongyu-lee committed:
feat: remove vite.svg icon file, change html head title, apply new design for password-related dialog

yoongyu-lee committed:
feat: removing TasProperties file

yoongyu-lee committed:
feat: remove unused html files, add sending entity info api(temporary)

yoongyu-lee committed:
feat: remove notification configuration files (yml, property)

yoongyu-lee committed:
add '*' to label of required fields

yoongyu-lee committed:
chore(yaml): reset value to default value

yoongyu-lee committed:
feat: enable actuator shutdown endpoint

yoongyu-lee committed:
feat: Fix bug in TA registration process

yoongyu-lee committed:
feat: add SpaController

yoongyu-lee committed:
fix: correct typos

yoongyu-lee committed:
chore: reset yaml files to default

yoongyu-lee committed:
chore: update latest frontend build files

yoongyu-lee committed:
fix: properly shutdown ScheduledExecutorService on application exit

Added a @PreDestroy method to ensure the ScheduledExecutorService is properly shut down when the application stops. This prevents lingering non-daemon threads from keeping the process alive after shutdown.

yoongyu-lee committed:
chore: reset yaml file to default

yoongyu-lee committed:
feat: update directory paths for simple install (TAS → TA, CAS → CA)

yoongyu-lee committed:
fix: align session state type with ExtendedSession

yoongyu-lee committed:
feat: add request-ta-did API to List Provider

yoongyu-lee committed:
feat: display admin type text below the title image

yoongyu-lee committed:
feat: remove frontend files from backend static dir, integrate frontend build in build.gradle

yoongyu-lee committed:
refactor: rename ts filenames to start with lowercase

yoongyu-lee committed:
docs: WIP on admin guide

yoongyu-lee committed:
feat: Complete Korean version of Admin manual and fix bugs found during writing

yoongyu-lee committed:
chore: add frontend dependency licenses and license generation script

yoongyu-lee committed:
chore(readme): update readme files

yoongyu-lee committed:
docs: update TA server and admin guide

yoongyu-lee committed:
docs: update changelog, license info, and frontend/backend README files

yoongyu-lee committed:
docs(license): add LICENSE to frontend for making dependencies-license.

yoongyu-lee committed:
docs(frontend-readme): update frontend readme.md

yoongyu-lee committed:
docs(readme): updated readme

yoongyu-lee committed:
docs(frontend-readme): updated frontend readme

yoongyu-lee committed:
docs(frontend-readme): updated frontend readme

yoongyu-lee committed:
docs: remove LICENSE in frontend directory

yoongyu-lee committed:
fix: resolve static analysis issues (resource leak, thread safety, exception handling)

yoongyu-lee committed:
feat(build): fix bean loading issue in 'sample' profile for tests

- Exclude ContractApiConfig and BlockChainServiceImpl when 'sample' profile is active
- Add MockStorageServiceImpl for satisfying StorageService dependency in sample profile
- Ensures tests can run without blockchain configuration

yoongyu-lee committed:
docs(api): update API docs for v1.0.1 (dev)

yoongyu-lee committed:
docs(readme): fix typo

yoongyu-lee committed:
docs(table-definition): update table-definition for v1.0.1 (dev)

yoongyu-lee committed:
docs(guide): update 'open did installation guide' link

yoongyu-lee committed:
docs(installation-guide): fix word TAS -> TA

yoongyu-lee committed:
feat: update init data to list_vc_plan (for 1dev, for demo)

yoongyu-lee committed:
docs(table-definition): update erd-cloud link

jinhwankim6557 reviewed (approved):
I'm really looking forward to the Admin Console feature. :)

yoongyu-lee committed:
temp: temp rename for casing fix

yoongyu-lee committed:
fix: Fix filename casing to match imports (Api.ts, Regex.ts)

gw-nam committed:
chore: Modify Github Action

yoongyu-lee committed:
feat: make vc-schema API params optional and manage certificate schema via separate file

- Changed `id` and `name` parameters of vc-schema API to optional
- Temporarily manage certificate VC schema as a separate static file
  (to be replaced with DB management by June)

yoongyu-lee committed:
fix: fix certificate-vc url

jinhwankim6557 reviewed (approved):
LGTM :)
